.github/workflows/go-ci.yaml

```diff
@@ -0,0 +1,195 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+---
+name: Go Continuous Integration
+
+on:
+  workflow_call:
+    secrets:
+      gh_username:
+        required: false
+      gh_token:
+        required: false
+    inputs:
+      go-version:
+        description: "The version of Go to use"
+        required: false
+        type: string
+      go-private:
+        description: "The GOPRIVATE environment variable"
+        required: false
+        type: string
+      run-benchmarks:
+        description: "Run benchmarks"
+        required: false
+        default: false
+        type: boolean
+      cache:
+        description: "Use caching"
+        required: false
+        default: false
+        type: boolean
+      run-vuln-scan:
+        description: "Scan for vulnerabilities (using Trivy)"
+        required: false
+        default: true
+        type: boolean
+      run-release-test:
+        description: "Run release dry-run using GoReleaser"
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  tests:
+    env:
+      GOPRIVATE: "${{ inputs.go-private }}"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        if: inputs.go-version != ''
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ inputs.go-version }}
+
+      - name: Setup Go
+        if: inputs.go-version == ''
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Setup private repository access
+        if: inputs.go-private != ''
+        run: git config --global url."https://${{ secrets.gh_username }}:${{ secrets.gh_token }}@github.com".insteadOf "https://github.com"
+
+      - name: Check go mod tidy
+        run: |
+          go mod tidy
+          if git diff --exit-code go.mod go.sum; then
+            echo "go.mod and go.sum are tidy."
+          else
+            echo "::error go.mod or go.sum are not tidy. Please run 'go mod tidy'."
+            exit 1
+          fi
+
+      - name: Restore Go cache
```

Comment:
Shouldn't the `setup-go` action be able to do necessary caching? https://github.com/actions/setup-go?tab=readme-ov-file#caching-dependency-files-and-build-outputs